apps/jubmoji-quest/src/pages/cardholder/index.tsx

```diff
@@ -0,0 +1,486 @@
+                <PowerCard
+                  title="Propose a quest"
+                  description="Cardholders can propose quests for the community"
+                  powerType="REDIRECT"
+                />
+              </div>
+
+              <div
+                onClick={() =>
+                  setCardholderAction(CardholderActions.JOIN_TG_GROUP)
+                }
+              >
+                <PowerCard
+                  title="Join cardholder TG group"
+                  description="Group where updates are sent & feedback can be shared"
+                  powerType="TELEGRAM"
+                />
+              </div>
+
+              <Message>{`Open to all`}</Message>
+
+              <Link href="/card-request">
+                <PowerCard
+                  title="Apply to be a cardholder"
+                  description="Describe what you'd like to do with your card during Devconnect"
+                  powerType="QR_CODE"
+                />
+              </Link>
+            </div>
+          </div>
+        </div>
+      </div>
+    </>
+  );
+}
+
+enum CardholderTapModalState {
+  UNLOCKING,
+  EDITING,
+  CONFIRMING,
+}
+
+interface CardholderTapModalProps extends ModalProps {
+  action: CardholderActions;
+}
+
+const CardholderTapModal = ({
+  action,
+  isOpen,
+  setIsOpen,
+}: CardholderTapModalProps) => {
+  const md = useRef<any>();
+  useEffect(() => {
+    md.current = new MobileDetect(window?.navigator?.userAgent);
+  }, []);
+  const deviceImage = md?.current?.is("iPhone")
+    ? "/images/tap-phone-ios.png"
+    : "/images/tap-phone-android.png";
+
+  const [modalState, setModalState] = useState(
+    CardholderTapModalState.UNLOCKING
+  );
+  const [pubKeyIndex, setPubKeyIndex] = useState<number>();
+  const [meaning, setMeaning] = useState<string>();
+  const [telegramLink, setTelegramLink] = useState<string>();
+  const [website, setWebsite] = useState<string>("");
+
+  const initialTap = async () => {
+    const initMessage = getRandomNullifierRandomness();
+    const messageHash = bigIntToHex(getMessageHash(initMessage));
+    let command = {
+      name: "sign",
+      keyNo: 1,
+      digest: messageHash,
+    };
+
+    let res: {
+      input: { digest: string };
+      signature: { raw: NfcCardRawSignature };
+      publicKey: string;
+    };
+    try {
+      // --- request NFC command execution ---
+      res = await execHaloCmdWeb(command, {
+        statusCallback: (cause: any) => {
+          if (cause === "retry") {
+            toast.error("Tapping failed, please try again.");
+          } else {
+            console.log("Tapping status", cause);
+          }
+        },
+      });
+
+      const proofInstance = createProofInstance(PublicMessageSignature, {
+        randStr: "",
+      });
+      const pubKeyIndex = cardPubKeys.findIndex(
+        (key) => key.pubKeySlot1 === res.publicKey // Use secp256k1 key here, which is in slot 1
+      );
+      const proof = await proofInstance.prove({
```

Comment:
Why do we need to prove and verify here?

Reply:
Idk just to do some error handling, but ill remove